### ISSUE TITLE
capriqorn sasmol implementation

### ISSUE THREAD

**Issue by madscatt** [madscatt/zazzie#14]
Currently, capriqorn now has a separate python modules to run

    cadishi/	capriqorn/

these are installed and run okay on entropy in  ~/.local

    ~/.local/bin/generate_preprocessor_input
    ~/.local/bin/preprocessor
    ~/.local/bin/generate_histograms_input
    ~/.local/bin/histograms_par
    ~/.local/bin/generate_postprocessor_input
    ~/.local/bin/postprocessor

    #(edit remove ADAPT_INPUT_PATH) in preprocessor.yaml & postprocessor.yaml 

There are a few minor issues to perhaps deal with ADAPT_INPUT_PATH at a couple stages.

Installing this on Mac (Anaconda 4.1.1 (x86_64)) does not compile correctly.  Both the cuda bits and the capriqorn bits failed to compile correctly.

    Processing dependencies for capriqorn==0.1
    Searching for disthist
    Reading http://pypi.python.org/simple/disthist/
    Couldn't find index page for 'disthist' (maybe misspelled?)
    Scanning index of all packages (this may take a while)
    Reading http://pypi.python.org/simple/
    No local packages or download links found for disthist
    error: Could not find suitable distribution for Requirement.parse('disthist')`

This is after adding a symbolic link 

    /Developer/NVIDIA/CUDA-6.5/lib --> /Developer/NVIDIA/CUDA-6.5/lib64

(lib existed, lib64 does not on El Capitan)

     1 warning and 18 errors generated.
    error: command '/Developer/NVIDIA/CUDA-6.5/bin/nvcc' failed with exit status 1

will not finish compiling and disthist is not available for capriqorn to install.

**Comment by madscatt** [comment]
Capriqorn Python 3 port summary:

- Restored Capriqorn from the April 2016 upstream source archive and legacy SASSIE wrapper state.
- Ported the Capriqorn workflow to Python 3 while preserving the legacy module flow:
  `main() -> unpack_variables() -> setup_logging() -> general_setup() -> initialization() -> run() -> epilogue()`.
- Added/updated the active module under:
  - `sassie.calculate.capriqorn`
  - `sassie.interface.capriqorn`
  - `gui_mimic_capriqorn.py`
- Replaced external script/executable calls with direct in-package Python calls for:
  - YAML generation
  - preprocessor
  - histogram driver
  - postprocessor
- Restored package-owned upstream dependencies under Capriqorn:
  - `capriqorn`
  - `disthist`
  - `pydh`
  - `pydh_knl`
  - optional `cudh`
  - optional `cudh_knl`
- Built `pydh_knl` through the repo-level `setup.py`.
- Added CUDA build support gated by:
  `SASSIE_ENABLE_CAPRIQORN_CUDA=1`
- Implemented automatic runtime backend selection:
  - CUDA GPU is used when installed and a runtime probe succeeds.
  - CPU `pydh` fallback is used otherwise.
  - There is no GUI/user backend selector.
- Added visible runtime backend reporting:
  - `Using Capriqorn backend: CUDA GPU`
  - `Using Capriqorn backend: CPU`
- Added an installed-package smoke helper:
  `python -m sassie.calculate.capriqorn.installed_smoke_capriqorn`
- Validated clean installed CUDA and CPU-fallback behavior on `zazzie.genapp.rocks`.
- Added targeted tests for:
  - filter validation
  - seeded gui mimic smoke
  - dependency/build guard behavior
  - automatic CPU/GPU YAML selection
  - optional CUDA histogram parity
  - optional lysozyme CPU/GPU workflow equivalence

Current validation status:

- Local targeted Capriqorn tests: `16 passed, 2 skipped`
- Server targeted Capriqorn tests with CUDA available: `16 passed, 2 skipped`
- Installed CUDA smoke on `zazzie.genapp.rocks`: passed, selected CUDA
- Installed CPU fallback smoke with `CUDA_VISIBLE_DEVICES=-1`: passed, selected CPU
- Lysozyme CPU/GPU double-precision workflow comparison: exact match for `Iq.txt` and `10/intensity/dI`

Guardrails / remaining scope:

- Capriqorn is currently validated for the historical lysozyme DCD/XST lane.
- Broader non-lysozyme input coverage is not yet claimed.
- CUDA is validated on `zazzie.genapp.rocks`; Apple Metal is a future backend evaluation.
- DTYPE/precision policy remains deferred.
- Scientific algorithms, output formats, and ordering were preserved rather than refactored.

```text
Capriqorn was ported into the Python3 GenApp driver layer using the current SASSIE gui_mimic_capriqorn.py as the naming and usage reference.

The new GenApp bin driver is bin/capriqorn. It imports the current SASSIE Capriqorn interface and runner:

sassie.interface.capriqorn.capriqorn_filter
sassie.calculate.capriqorn.capriqorn

The web/module inputs now follow the SASSIE interface names:

runname
pdbfile
dcdfile
xstfile
aliasfile
number_q_values
q_max
create_alias_flag

The module JSON was updated to expose the current workflow fields. The reference PDB, trajectory file, XST box-size file, and alias file are separate inputs. Local testing defaults were added under bin/local_data_for_testing/capriqorn using the current Capriqorn test-data set.

create_alias_flag is exposed under Advanced Input and defaults to off. Its help text notes that it is intended to let SASSIE generate alias.dat from the PDB atom names. At the time of the GenApp port, the SASSIE-side create_alias_file() path is still placeholder/stub behavior, so the default web workflow continues to require an aliasfile.

The driver preserves the standard GenApp JSON execution path and adds the usual --local command-line mode for development testing. It reports progress through SASSIE STATUS messages and returns a concise textarea summary rather than dumping backend chatter.

After completion, the driver reads:

<runname>/capriqorn/Iq.txt

and returns a Plotly I(q) plot. The summary textarea lists key generated files, including Iq.txt, intensities.h5, and the generated YAML files.

Capriqorn was moved out of the restricted alpha menu and into the Calculate section.

Local verification completed:

python -m py_compile bin/capriqorn
PYTHONPATH=/Users/curtisj/git_working_copies/zazzie/src python bin/capriqorn --local

The local smoke test completed successfully against the current SASSIE source tree and returned a valid Plotly I(q) object with progress at 100%.

The server environment was checked separately and now has sassie.calculate.capriqorn installed in site-packages, so the earlier limitation that the local installed SASSIE package lacked Capriqorn is resolved on zazzie.genapp.rocks.

Commit:

6d09483 Port Capriqorn GenApp driver
```
